### PR TITLE
feat: get creating, updating, deleting Integrations working

### PR DESCRIPTION
**Pull request by tomkit**
- Prisma migration for `Integration`: adds `is_enabled` and makes `config` optional
- Use Nextjs `useSWR`'s `mutate` to fetch and make updates
- Get creating, updating, deleting Integrations working
- Update openapi spec for creating/updating integrations

SWR related reading for using mutating local and remote state:
- https://github.com/vercel/swr/issues/157
- https://github.com/vercel/swr/pull/1745
- https://github.com/vercel/swr-site/pull/202/files

Other:
- Remove extra /dashboard page, use root instead for dashboard
- Use `Card` for `MetricCard` rather than `Grid`
- Rename `Active` tab to `Installed`
- Rename `Application 1` to `Your application`
- others

### CODE REVIEW

File: apps/mgmt-ui/src/components/configuration/IntegrationCard.tsx [hunk 1]
```diff
@@ -24,12 +24,38 @@ export default function IntegrationCard(props: {
...
+                    authType: 'oauth2',
+                    category,
+                    providerName,
+                    isEnabled: true, // TODO: we need another notion of live vs enabled
+                    applicationId: APPLICATION_ID,
+                  };
+                  const updatedIntegrations = [...existingIntegrations, newIntegration];
+
+                  mutate(updatedIntegrations, false);
+                  mutateIntegration(createRemoteIntegration(newIntegration), false);
```
Comment (tomkit) [inline apps/mgmt-ui/src/components/configuration/IntegrationCard.tsx:48]: Following the pattern from the linked SWR reading for now

File: apps/mgmt-ui/src/components/configuration/VerticalTabs.tsx [hunk 2]
```diff
@@ -119,8 +123,8 @@ type ConfigurationTab = {
 };
 const configurationTabs: ConfigurationTab[] = [
   {
-    label: 'Active',
-    value: 'active',
+    label: 'Installed',
+    value: 'installed',
```
Comment (tomkit) [inline apps/mgmt-ui/src/components/configuration/VerticalTabs.tsx:127]: Renaming to Installed since you can have existing but disabled integrations

File: apps/mgmt-ui/src/components/customers/MetricCard.tsx [hunk 3]
```diff
@@ -10,37 +10,15 @@ export default function MetricCard(props: MetricCardProps) {
...
-            root: 'my-4 mr-4',
-          }}
-        >
-          <Typography variant="subtitle1" noWrap>
-            {title}
-          </Typography>
-          <Typography variant="body2" noWrap>
+    <Card variant="outlined" className="h-32">
+      <Stack>
+        <CardHeader avatar={icon} subheader={title} />
```
Comment (tomkit) [inline apps/mgmt-ui/src/components/customers/MetricCard.tsx:15]: Make use of mui's `CardHeader`, `CardContent` so you get responsive for free

File: apps/mgmt-ui/src/components/logs/ModelSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        <Select labelId="model-select-label" id="model-select" value={method} label="Model" onChange={handleChange}>
+          <MenuItem value="account">Account</MenuItem>
+          <MenuItem value="contact">Contact</MenuItem>
+          <MenuItem value="lead">Lead</MenuItem>
+          <MenuItem value="opportunity">Opportunity</MenuItem>
+        </Select>
+      </FormControl>
+    </Box>
+  );
+}
```
Comment (tomkit) [inline apps/mgmt-ui/src/components/logs/ModelSelect.tsx:28]: Used on sync log page

File: apps/mgmt-ui/src/pages/index.tsx [hunk 5]
```diff
@@ -23,19 +29,83 @@ export default function Home() {
...
+                      <Card>
+                        <CardContent>Code block 2</CardContent>
+                      </Card>
+                    </Grid>
+                  </Grid>
+                </Grid>
+              </Grid>
+            </CardContent>
+          </Card>
+        </Box>
```
Comment (tomkit) [inline apps/mgmt-ui/src/pages/index.tsx:108]: Moved from the dashboard page

File: openapi/mgmt/components/schemas/objects/integration.yaml [hunk 6]
```diff
@@ -16,30 +16,34 @@ properties:
...
-        - crm.objects.deals.read
-        - crm.objects.contacts.write
-        - crm.objects.companies.write
-        - crm.objects.deals.write
-        credentials:
-        - oauth_client_id: 7393b5a4-5e20-4648-87af-b7b297793fd1
-          oauth_client_secret: 941b846a-5a8c-48b8-b0e1-41b6d4bc4f1a
-    - sync:
-      - period_ms: 60000
+      - remote_provider_app_id: my_app_id
```
Comment (tomkit) [inline openapi/mgmt/components/schemas/objects/integration.yaml:35]: indentation

File: apps/mgmt-ui/src/client.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
+import { snakecaseKeys } from './utils/snakecase';
+
+export const API_HOST = 'http://localhost:8080';
+
+// TODO: get this on the server-side from the session
+export const APPLICATION_ID = 'a4398523-03a2-42dd-9681-c91e3e2efaf4';
+
+// TODO: use Supaglue TS client
+export async function updateRemoteIntegration(data: any) {
```
Comment (asdfryan) [inline apps/mgmt-ui/src/client.ts:9]: data can be typed

File: apps/mgmt-ui/src/client.ts [hunk 8]
```diff
@@ -0,0 +1,49 @@
+import { snakecaseKeys } from './utils/snakecase';
+
+export const API_HOST = 'http://localhost:8080';
+
+// TODO: get this on the server-side from the session
+export const APPLICATION_ID = 'a4398523-03a2-42dd-9681-c91e3e2efaf4';
+
+// TODO: use Supaglue TS client
```
Comment (asdfryan) [inline apps/mgmt-ui/src/client.ts:8]: Most of this file can be DRYed

File: apps/mgmt-ui/src/components/configuration/IntegrationDetailTabPanel.tsx [hunk 9]
```diff
@@ -72,29 +83,62 @@ export default function IntegrationDetailTabPanel(props: IntegrationDetailTabPan
...
+      <Stack direction="row" className="gap-2 justify-between">
+        <Stack direction="row" className="gap-2">
+          <Button
+            variant="outlined"
+            onClick={() => {
+              router.back();
+            }}
+          >
+            Cancel
+          </Button>{' '}
```
Comment (asdfryan) [inline apps/mgmt-ui/src/components/configuration/IntegrationDetailTabPanel.tsx:95]: Remove `{' '}`

File: apps/mgmt-ui/src/components/configuration/IntegrationDetailTabPanel.tsx [hunk 10]
```diff
@@ -72,29 +83,62 @@ export default function IntegrationDetailTabPanel(props: IntegrationDetailTabPan
...
+          <Button
+            variant="contained"
+            onClick={() => {
+              const newIntegration = {
+                ...integration,
+                config: {
+                  providerAppId: '',
```
Comment (asdfryan) [inline apps/mgmt-ui/src/components/configuration/IntegrationDetailTabPanel.tsx:102]: Why is this line necessary? Does it get overridden by the next line?

File: apps/mgmt-ui/src/components/configuration/IntegrationDetailTabPanel.tsx [hunk 11]
```diff
@@ -72,29 +83,62 @@ export default function IntegrationDetailTabPanel(props: IntegrationDetailTabPan
...
+                  oauth: {
+                    ...integration?.config?.oauth,
+                    credentials: {
+                      oauthClientId: clientId,
+                      oauthClientSecret: clientSecret,
+                    },
+                    oauthScopes: oauthScopes.split(','),
+                  },
+                  sync: {
+                    periodMs: 60 * 60 * 1000,
```
Comment (asdfryan) [inline apps/mgmt-ui/src/components/configuration/IntegrationDetailTabPanel.tsx:113]: Is 1x/hour what we want?

File: apps/mgmt-ui/src/components/configuration/IntegrationDetailTabPanel.tsx [hunk 12]
```diff
@@ -29,9 +33,16 @@ export default function IntegrationDetailTabPanel(props: IntegrationDetailTabPan
   return (
     <Stack direction="column" className="gap-4">
       <Stack direction="row" className="items-center justify-between w-full">
-        <Stack direction="row">
-          {providerToIcon(integrationCardInfo.providerName)}
-          <Typography variant="subtitle1">{integrationCardInfo.name}</Typography>
+        <Stack direction="row" className="items-center justify-center gap-2">
+          {providerToIcon(integrationCardInfo.providerName, 35)}
```
Comment (asdfryan) [inline apps/mgmt-ui/src/components/configuration/IntegrationDetailTabPanel.tsx:37]: ICON_SIZE?

File: apps/mgmt-ui/src/components/configuration/IntegrationDetailTabPanel.tsx [hunk 13]
```diff
@@ -17,8 +19,10 @@ export default function IntegrationDetailTabPanel(props: IntegrationDetailTabPan
...
   const [clientSecret, setClientSecret] = useState('');
   const [oauthScopes, setOauthScopes] = useState('');
+  const router = useRouter();
 
-  const { trigger } = useSWRMutation('/mgmt/v1/integrations', sendRequest);
+  const { mutate: mutateIntegration } = useIntegration(integration?.id);
+  const { integrations: existingIntegrations = [], mutate } = useIntegrations();
 
   useEffect(() => {
     setClientId(integration?.config?.oauth?.credentials?.oauthClientId);
```
Comment (asdfryan) [inline apps/mgmt-ui/src/components/configuration/IntegrationDetailTabPanel.tsx:28]: Looks like integration is already passed in as a prop -- why do we need to do this?

File: apps/mgmt-ui/src/components/configuration/IntegrationDetailTabPanel.tsx (hunk 11 above)
Comment (tomkit) [reply apps/mgmt-ui/src/components/configuration/IntegrationDetailTabPanel.tsx:113]: For now

**Review by asdfryan — APPROVED** [review]
